tests/fetchhead/nonetwork.c

```diff
@@ -406,3 +406,64 @@ void test_fetchhead_nonetwork__create_when_refpecs_given(void)
 	git_remote_free(remote);
 	git_buf_free(&path);
 }
+
+static int nrefs_found;
+static bool count_refs_called;
+
+int count_refs(const char *ref_name, const char *remote_url, const git_oid *oid, unsigned int is_merge, void *payload)
+{
+	const char *pfix = (const char *) payload;
+
+	GIT_UNUSED(remote_url);
+	GIT_UNUSED(oid);
+	GIT_UNUSED(is_merge);
+
+	count_refs_called = true;
+
+
+	if (!git__prefixcmp(ref_name, pfix))
+		nrefs_found++;
+
+	return 0;
+}
+
+void test_fetchhead_nonetwork__create_with_multiple_refspecs(void)
+{
+	git_remote *remote;
+	git_buf path = GIT_BUF_INIT;
+
+	cl_set_cleanup(&cleanup_repository, "./test1");
+	cl_git_pass(git_repository_init(&g_repo, "./test1", 0));
+
+	cl_git_pass(git_buf_joinpath(&path, git_repository_path(g_repo), "FETCH_HEAD"));
+	cl_git_pass(git_remote_create(&remote, g_repo, "origin", cl_fixture("testrepo.git")));
+	git_remote_free(remote);
+	cl_git_pass(git_remote_add_fetch(g_repo, "origin", "+refs/notes/*:refs/origin/notes/*"));
+	/* Pick up the new refspec */
+	cl_git_pass(git_remote_lookup(&remote, g_repo, "origin"));
+
+	cl_assert(!git_path_exists(path.ptr));
+	cl_git_pass(git_remote_fetch(remote, NULL, NULL, NULL));
+	cl_assert(git_path_exists(path.ptr));
+
+	count_refs_called = 0;
+	nrefs_found = 0;
+	cl_git_pass(git_repository_fetchhead_foreach(g_repo, count_refs, "refs/notes/"));
```

Comment:
Instead of calling this function trice, you could just:

```
struct fetchhead_foreach_payload {
    int refs_notes_found;
    int refs_heads_found;
    int refs_tags_found;
}

int count_refs(..., void *payload)
{
    struct fetchhead_foreach_payload *f = (...) payload;

    if (!git__prefixcmp("refs/notes/", pfix))
        f->refs_notes_found++;
   ....
}
```

Reply:
I didn't want to make this function to tightly coupled with these particular prefixes we want to count right now. I'll see if passing prefix-count tuples in ends up being less verbose.

Reply:
I've just pushed up a new commit which just runs it the one time without coupling it to the specific prefixes we happen to use right now. It doesn't save too much space, but I think it does show the intent more clearly.